novawallet/Common/Services/ExtrinsicService/Substrate/FeeManaging/Hydra/HydraExtrinsicFeeInstaller.swift

```diff
@@ -42,20 +47,19 @@ extension HydraExtrinsicFeeInstaller: ExtrinsicFeeInstalling {
     }
 
     private func createTransferFeeCalls(using assetId: HydraDx.LocalRemoteAssetId) -> TransferFeeInstallingCalls {
-        let shouldSetCurrency = feeAsset.asset.assetId != HydraDx.nativeAssetId
-
-        let (setCurrencyCall, revertCurrencyCall): (HydraDx.SetCurrencyCall?, HydraDx.SetCurrencyCall?) = {
-            guard shouldSetCurrency else { return (nil, nil) }
-
-            return (
-                .init(currency: assetId.remoteAssetId),
-                .init(currency: HydraDx.nativeAssetId)
+        guard
+            let currentFeeAssetId = swapState.feeCurrency,
```

Comment:
The logic must be the following: 
1) if current fee asset is different from new fee asset then insert call setCurrency(newFeeAsset) in the beginning;

2) if the fee asset after the operation is not native then include setCurrency(nativeAsset) to the end;